test/functional/test_framework/util.py

```diff
@@ -296,16 +296,28 @@ def initialize_datadir(dirname, n, chain):
     datadir = get_datadir_path(dirname, n)
     if not os.path.isdir(datadir):
         os.makedirs(datadir)
+    write_config(os.path.join(datadir, "bitcoin.conf"), n, chain)
+    os.makedirs(os.path.join(datadir, 'stderr'), exist_ok=True)
+    os.makedirs(os.path.join(datadir, 'stdout'), exist_ok=True)
+    return datadir
+
+
+def write_config(config_path, n, chain, extra_config=""):
     # Translate chain name to config name
     if chain == 'testnet3':
         chain_name_conf_arg = 'testnet'
         chain_name_conf_section = 'test'
+    elif chain == "main":
```

Comment:
Thanks, added second commit with suggested cleanups

Reply:
re: https://github.com/bitcoin/bitcoin/pull/17556#discussion_r377285894

> Thanks, added second commit with suggested cleanups

Had to drop second commit because it was causing feature_backwards_compatibility.py failures. (Previous versions require `regtest=1` not `chain=regtest`)